src/datachain/lib/dc.py

```diff
@@ -1583,15 +1635,18 @@ def from_records(
             ```
         """
         session = Session.get(session, in_memory=in_memory)
-        catalog = session.catalog
+        catalog = kwargs.get("catalog") or session.catalog
```

Comment:
do we pass catalog anywhere directly? why do we need this change?

Reply:
This is a leftover, I removed it